src/satosa/base.py

```diff
@@ -134,6 +134,12 @@ def _auth_resp_callback_func(self, context, internal_response):
         """
 
         context.request = None
+        context_state = context.state.get(STATE_KEY)
+        if not context_state:
+            redirect_url = self.config.get("UNKNOW_ERROR_REDIRECT_PAGE")
+            raise SATOSAStateError(('context.state has no {}. Your session '
+                                    'is not valid, please start a new '
+                                    'Authentication request again.'.format(STATE_KEY)))
```

Comment:
I think it is good to catch such cases or exceptions, but first we should make sure we understand the error, where it comes from and why it is happening. Can we change the code to make it as if the error did not happen at all?

Reply:
As @ioparaskev said, SATOSA is currently not concerned about user-facing messages. This message is a mix of technical ("context.state has no") and user-facing information ("please start a new authentication"). I think that neither part is good. What we want to provide in an exception is not a message, but error context. Which information is useful to understand why this happened? This is the information that the exception should hold. I lean on making this information structured, a dict with meaningful keys and data as values.

Reply:
I think that the PR suggested by @skoranda is the solution for the 60% of this kind of problem, really. I follow the users in their daily abits and many of them use the Disco url to bookmark the SP... That's it.

Scott gave us the remediation here:
https://github.com/IdentityPython/SATOSA/pull/322

Move in that direction, merge that first then come back here with better refactoring. I'm behind you (but with 2 meters of covid-social-distancing!)

Reply:
ping here